gobblin-runtime/src/main/java/org/apache/gobblin/runtime/ThrowableWithErrorCode.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.runtime;
+
+public interface ThrowableWithErrorCode {
+
+  /**
+   * Returns an error code that uniquely identifies the problem in the given context.
+   *
+   * The error code will be used programmatically, to take different recovery actions.
+   * */
+  String getErrorCode();
```

Comment:
Maybe in the javadoc include a couple of example error codes, from this it's not quite clear if an error code would be just a number, a single word, a short description of the issue, etc. I guess any of those would work the same, but would probably be good to have a standard.

Reply:
Added an example.